modules/ebs_snapshot_explorer/main.py

```diff
@@ -0,0 +1,303 @@
+    """Deletes a given list of volumes
+
+    If the volume is in use, the volume is forcibly detached because this module
+    only deals with temporary copies so data integrity is not a high priority when
+    a volume is ready to be detatched. After the volume is forcibly detatched, the
+    volume will be deleted after the detaching operation finishes.
+    """
+    failed_volumes = []
+    for volume in volumes:
+        try:
+            client.delete_volume(VolumeId=volume)
+        except ClientError as error:
+            code = error.response['Error']['Code']
+            if code == 'VolumeInUse':
+                client.detach_volume(
+                    VolumeId=volume,
+                    Force=True)
+                waiter = client.get_waiter('volume_available')
+                waiter.wait(VolumeIds=[volume])
+                client.delete_volume(VolumeId=volume)
+                continue
+            failed_volumes.append(volume)
+    return failed_volumes
+
+
+def delete_snapshots(client, snapshots):
+    """Deletes a given list of snapshots"""
+    failed_snapshots = []
+    for snapshot in snapshots:
+        try:
+            client.delete_snapshot(SnapshotId=snapshot)
+        except ClientError:
+            failed_snapshots.append(snapshot)
+    return failed_snapshots
+
+
+def cleanup(client):
+    """Cleans up the temporary snapshots and volumes created during this
+    modules execution
+    """
+    new_data = {}
+    success = True
+    temp_file = Path(__file__).parent / 'temp.json'
+    if temp_file.is_file():
+        with temp_file.open('r') as file:
+            data = json.load(file)
+            if 'snapshots' in data:
+                new_data['snapshots'] = delete_snapshots(client, data['snapshots'])
+            if 'volumes' in data:
+                new_data['volumes'] = delete_volumes(client, data['volumes'])
+        if 'volumes' in new_data and new_data['volumes']:
+            print('  Failed to delete volumes: {}'.format(new_data['volumes']))
+            success = False
+        if 'snapshots' in new_data and new_data['snapshots']:
+            print('  Failed to delete snapshots: {}'.format(new_data['snapshots']))
+            success = False
+        store_temp_data(new_data)
+        if success:
+            temp_file.unlink()
+    return success
+
+
+def store_temp_data(data):
+    """Stores temporary data in a JSON file"""
+    temp_file = Path(__file__).parent / 'temp.json'
+    if temp_file.exists():
+        with temp_file.open('r') as json_file:
+            existing_data = json.load(json_file)
+            data.update(existing_data)
+    with temp_file.open('w+') as json_file:
+        json.dump(data, json_file)
+
+
+def main(args, pacu):
+    """Main module function, called from Pacu"""
+    args = parser.parse_args(args)
+    session = pacu.get_active_session()
+    print = pacu.print
+
+    instance = args.instance
+    region = args.region
+    zone = region + args.zone
```

Comment:
This should be a little more clear in the args. I was thinking of zone as `us-east-1a` instead of just `a`